### PR TITLE
Exploding

### PR DESCRIPTION
**Pull request by eric-s-s**
comment, oscar

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -333,6 +340,34 @@ Modifier
 
     - .get_modifier()
 
+Exploding
+    A die that continues rolling if the maximum value is rolled, adding the previous roll to the results.
+    It rolls up to (explosions) number of times. So, dt.Exploding(dt.Die(6), explosions=3) will keep rolling
+    as long as a 6 was rolled, to a maximum of 3 re-rolls, that means it rolls, 1-5, then 7-11 (6 + 1to5),
+    then 13-17, and finally, 19-24. Any modifiers and multipliers are applied to each re-roll.
+    dt.Exploding(dt.ModDie(6, 1), explosions=2) rolls, 2-6, then 9-13, and finally, 15-21.
+
```
Comment (Oscar-Lin) [inline README.rst]: quick

File: README.rst [hunk 2]
```diff
@@ -333,6 +340,34 @@ Modifier
 
     - .get_modifier()
 
+Exploding
+    A die that continues rolling if the maximum value is rolled, adding the previous roll to the results.
```
Comment (Fysty) [inline README.rst]: too many rolls

File: README.rst [hunk 3]
```diff
@@ -333,6 +340,34 @@ Modifier
...
+    It rolls up to (explosions) number of times. So, dt.Exploding(dt.Die(6), explosions=3) will keep rolling
+    as long as a 6 was rolled, to a maximum of 3 re-rolls, that means it rolls, 1-5, then 7-11 (6 + 1to5),
+    then 13-17, and finally, 19-24. Any modifiers and multipliers are applied to each re-roll.
+    dt.Exploding(dt.ModDie(6, 1), explosions=2) rolls, 2-6, then 9-13, and finally, 15-21.
```
Comment (Fysty) [inline README.rst]: should be 16~21

File: README.rst (hunk 2 above)
Comment (Fysty) [reply README.rst]: not clear, too long

An exploding die is a die that has a chance to roll again.

**Pull request merged**